server/access/streamBackend.go

```diff
@@ -0,0 +1,221 @@
+	"context"
+	"errors"
+	"fmt"
+	"github.com/onflow/flow-emulator/emulator"
+	"github.com/onflow/flow-go/engine/access/state_stream"
+	"github.com/onflow/flow-go/engine/access/state_stream/backend"
+	"github.com/onflow/flow-go/engine/common/rpc"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/storage"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/utils/logging"
+	"github.com/rs/zerolog"
+	"time"
+)
+
+type StateStreamBackend struct {
+	blockchain *emulator.Blockchain
+
+	log            zerolog.Logger
+	sendTimeout    time.Duration
+	responseLimit  float64
+	sendBufferSize int
+
+	getExecutionData GetExecutionDataFunc
+	getStartHeight   GetStartHeightFunc
+}
+
+func NewStateStreamBackend(blockchain *emulator.Blockchain, log zerolog.Logger) *StateStreamBackend {
+	return &StateStreamBackend{
+		blockchain:       blockchain,
+		log:              log,
+		sendTimeout:      state_stream.DefaultSendTimeout,
+		responseLimit:    state_stream.DefaultResponseLimit,
+		sendBufferSize:   state_stream.DefaultSendBufferSize,
+		getExecutionData: getExecutionDataFunc(blockchain),
+		getStartHeight:   getStartHeightFunc(blockchain),
+	}
+}
+
+var _ state_stream.API = &StateStreamBackend{}
+
+func getStartHeightFunc(blockchain *emulator.Blockchain) GetStartHeightFunc {
+	return func(blockID flow.Identifier, height uint64) (uint64, error) {
+		block, err := blockchain.GetBlockByID(blockID)
+		if err == nil {
+			return block.Header.Height, nil
+		}
+		return height, nil
+	}
+}
+
+func getExecutionDataFunc(blockchain *emulator.Blockchain) GetExecutionDataFunc {
+	return func(_ context.Context, height uint64) (*execution_data.BlockExecutionDataEntity, error) {
+		block, err := blockchain.GetBlockByHeight(height)
+
+		if err != nil {
+			return nil, storage.ErrNotFound
+		}
+		chunks := make([]*execution_data.ChunkExecutionData, 0)
+
+		for _, collectionGuarantee := range block.Payload.Guarantees {
+			collection := &flow.Collection{}
+			lightCollection, err := blockchain.GetCollectionByID(collectionGuarantee.CollectionID)
+			if err != nil {
+				return nil, err
+			}
+			var events []flow.Event
+			var txResults []flow.LightTransactionResult
+
+			for _, transactionId := range lightCollection.Transactions {
+				tx, err := blockchain.GetTransaction(transactionId)
+				if err != nil {
+					return nil, err
+				}
+				collection.Transactions = append(collection.Transactions, tx)
+
+				txResult, err := blockchain.GetTransactionResult(transactionId)
+				if err != nil {
+					return nil, err
+				}
+				events = append(events, txResult.Events...)
+
+				var lightResult flow.LightTransactionResult
+				lightResult.TransactionID = txResult.TransactionID
+				lightResult.ComputationUsed = 0
+				lightResult.Failed = txResult.Status != 0
+				txResults = append(txResults, lightResult)
```

Comment:
nit change to inline:
```go
lightResult := flow.LightTransactionResult{
    TransactionID: txResult.TransactionID,
    Failed: txResult.ErrorMessage != "",
}
```
not so nit:
Failed should be determined by checking the presence of an error: `Failed: txResult.ErrorMessage != ""`.

Reply:
https://github.com/onflow/flow-emulator/pull/496/commits/46deb1901110a133cc1a5475b92ebf89706e2898